### PR TITLE
Fixes #633, auto correct width for index page fixed

### LINKED ISSUE #633: Make suggestion box responsive on the homepage.
This can issue can be referred to: 
- #604 
- #584 
- #524

### PR DESCRIPTION
**Pull request by Marauderer97**
Fixes issue #633 

Changes: auto correct width for index page

Demo Link: [https://marauderer97.github.io/susper.com/](https://marauderer97.github.io/susper.com/)

Screenshots for the change: 

image
image
@nikhilrayaprolu @harshit98 *ngIf did not load the autocomplete component until it had results, that is why we were unable to access it from .ts files previously(Refer discussion on #633 ). Since we use the emitter object (hidecomponent) I removed *ngIf and autocorrect still works correctly.Meanwhile, I can access the component in any .ts file without any problem. Correct me if I am wrong.

### CODE REVIEW

File: src/app/auto-complete/auto-complete.component.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<div class="suggestion-box" *ngIf="results">
+<div id="sb" class="suggestion-box">
```
Comment (harshit98) [inline src/app/auto-complete/auto-complete.component.html]: Use an `id` name which is meaningful and can be understood. :)

**Comment by Marauderer97** [comment]
@harshit98 Done!
@nikhilrayaprolu Please review

**Comment by harshit98** [comment]
@Marauderer97 Please improve the corners. They should be sharp instead of being curved. Compare it with the market leader.

screenshot from 2017-07-31 23-06-54

**Comment by Marauderer97** [comment]
@harshit98 Done 
image

**Comment by harshit98** [comment]
@Marauderer97 Check your screenshot. The line should be straight w.r.t search bar. Currently, it is aligned slightly towards the right.

**Comment by Marauderer97** [comment]
@harshit98 Thank you for the advice, but you do realize that this is not part of the issue description? The issue was that autocorrect was not correctly sized on index page? and I have made sure that the size is the  same on the index page as the search bar. This is an extension, for which a seperate issue should have been raised. I am solving it here, to avoid lengthening the process. I assure you I always first tend to my open PRs, only when I feel I have hit a roadblock, or that I have satisfactorily solved the issue I open new ones. Also, the issue was a high priority one, that is why I opened a fresh PR to solve it.

**Comment by Marauderer97** [comment]
image
@harshit98 Please check

File: src/app/search-bar/search-bar.component.ts [hunk 2]
```diff
@@ -113,6 +115,9 @@ export class SearchBarComponent implements OnInit, AfterViewInit {
 
   ngOnInit() {
     this.displayStatus = 'hidebox';
+    this.boxWidth = document.getElementById('nav-group').style.width.replace('px', '');
```
Comment (nikhilrayaprolu) [inline src/app/search-bar/search-bar.component.ts]: can't we do this with just CSS?

Comment (Marauderer97) [inline src/app/search-bar/search-bar.component.ts]: No, @nikhilrayaprolu we cannot, we need different values for index and results page. From css, we can set only one value.

Comment (nikhilrayaprolu) [inline src/app/search-bar/search-bar.component.ts]: in that case, I would prefer writing two different classes A and B in search bar component and use class A when it is in the index page and class B when it is in the results page.In this way, we could maintain styles from just CSS but change them dynamically using javascript.

**Comment by Marauderer97** [comment]
@nikhilrayaprolu Did it in a more simple way, hope this is better. We can't do it from .css since it is different for index and results page

**Comment by nikhilrayaprolu** [comment]
@Marauderer97 it is not how much simple the code is, we have to consider how much maintainable it will be in future. 
when I was working on speech UI, I found that Google attaches classes dynamically, not the element styles. In future, if a new contributor wants to change this value, he might not find which component line is making the change, but if we attach the classes he can directly know which class should be changed.

**Comment by nikhilrayaprolu** [comment]
I don't mean to use just CSS for this. write styles as classes in style sheets, but change the class that element uses dynamically in javascript based on the present page.

**Comment by Marauderer97** [comment]
@nikhilrayaprolu As of now, I have used the same method that we have used for the search bar in the index page, setting the suggestion box to a different width, suitable for the index page

**Comment by harshit98** [comment]
@nikhilrayaprolu Please review this PR so that we can get this ready for merge!

**Comment by nikhilrayaprolu** [comment]
@Marauderer97 yes, we have done that for search bar but we were not aware of a better way of building it then. Now we are aware of other way and which is being used by tech giants like google and maintainable so it is advisable to follow better maintainable standard ways of doing things.

**Comment by Marauderer97** [comment]
@nikhilrayaprolu Changed the id instead of the css directly as requested. Hope this is satisfactory :+1:

**Comment by Marauderer97** [comment]
Closing this PR and making a fresh one.

**Pull request closed without merging**